### PR TITLE
[metadata] Add connectors to suggestion API

### PR DESCRIPTION
**Pull request by romainr**
- [metadata] Add connectors to suggestion API
- [metadata] Fix python styling of client API

### CODE REVIEW

**Comment by romainr** [comment]
@JohanAhlen Would help a tun to get the `connector: {'dialect': 'hive', 'id': 1}` in JS `optimizer.fetchPopularity()`?

(`analyzeRisk()` already has notebook which has connector)

**Comment by JohanAhlen** [comment]
Which optimizer? Local strategy already has the connector (this.connector I believe), for the API one it can be set in the constructor if needed.

File: desktop/libs/metadata/src/metadata/optimizer/dummy_client.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def upload_status(self, workload_id):
     pass
 
 
-  def top_tables(self, workfloadId=None, database_name='default', page_size=1000, startingToken=None):
+  def top_tables(self, workfloadId=None, database_name='default', page_size=1000, startingToken=None, connector=None):
```
Comment (JohanAhlen) [inline desktop/libs/metadata/src/metadata/optimizer/dummy_client.py:49]: I see. Will add...

**Review by JohanAhlen — APPROVED** [review]

File: desktop/libs/metadata/src/metadata/optimizer/dummy_client.py (hunk 1 above)
Comment (romainr) [reply desktop/libs/metadata/src/metadata/optimizer/dummy_client.py:49]: :+1:

**Pull request merged**